### PR TITLE
Adding benchmarks to envelop encryption integration tests

### PR DESCRIPTION
**Pull request by immutableT**
**What this PR does / why we need it**:
Adding benchmarks for envelop encryption integration tests.
Allows to estimate how envelop encryption may impact the performance of KubeAPI server.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/integration/master/secrets_enveloping_test.go [hunk 1]
```diff
@@ -79,117 +83,203 @@ resources:
...
+	kubeAPIServer     kubeapiservertesting.TestServer
+	restClient        *kubernetes.Clientset
+	ns                *corev1.Namespace
+	secret            *corev1.Secret
 }
 
-func runEnvelopeTest(t *testing.T, unSealSecretFunc unSealSecret, transformerConfigYAML, expectedEnvelopePrefix string) {
-	transformerConfig := parseTransformerConfigOrDie(t, transformerConfigYAML)
+func newEnvelopeTest(l kubeapiservertesting.Logger, transformerConfigYAML string) (*envelopTest, error) {
+	var e envelopTest
```
Comment (mikedanese) [inline test/integration/master/secrets_enveloping_test.go]: make this a struct literal
```golang
e := envelopeTest{
  logger: l,
  transformerConfig: transformerConfigYAML,
  storageConfig: framework.SharedEtcd(),
}
```

File: test/integration/master/secrets_enveloping_test.go [hunk 2]
```diff
@@ -79,117 +83,203 @@ resources:
...
 
-	client, err := kubernetes.NewForConfig(kubeAPIServer.ClientConfig)
+	e.restClient, err = kubernetes.NewForConfig(e.kubeAPIServer.ClientConfig)
 	if err != nil {
-		t.Fatalf("error while creating client: %v", err)
+		return nil, fmt.Errorf("error while creating rest client: %v", err)
 	}
 
-	ns, err := createTestNamespace(client, testNamespace)
+	e.ns, err = createTestNamespace(e.restClient, testNamespace)
```
Comment (mikedanese) [inline test/integration/master/secrets_enveloping_test.go]: all these assingments can be moved into the if, e.g.
```
if e.ns, err = createTestNamespace(e.restClient, testNamespace); err != nil {
  return nil, err
}
```

File: test/integration/master/secrets_enveloping_test.go [hunk 3]
```diff
@@ -79,117 +83,203 @@ resources:
...
+}
+
+func BenchmarkAESCBCEnvelopeWrite(b *testing.B) {
+	runBenchmark(b, aesCBCConfigYAML)
+}
 
-	if _, err := encryptionConfig.Write([]byte(transformerConfig)); err != nil {
-		return nil, fmt.Errorf("error while writing encryption config: %v", err)
+func runBenchmark(b *testing.B, transformerConfig string) {
+	b.StopTimer()
```
Comment (mikedanese) [inline test/integration/master/secrets_enveloping_test.go]: why is this necessary?

**Review by mikedanese — COMMENTED** [review]
/ok-to-test
/approve
/lgtm

File: test/integration/master/secrets_enveloping_test.go (hunk 3 above)
Comment (immutableT) [reply test/integration/master/secrets_enveloping_test.go]: I assume that the question is: why b.StopTimer is necessary.
The idea is to start measurements after test KubeAPI server is online. In other words, the benchmarks should focus on the crypto operations that are triggered by test.benchmark.

File: test/integration/master/secrets_enveloping_test.go (hunk 2 above)
Comment (immutableT) [reply test/integration/master/secrets_enveloping_test.go]: Done.

File: test/integration/master/secrets_enveloping_test.go (hunk 1 above)
Comment (immutableT) [reply test/integration/master/secrets_enveloping_test.go]: Done.

**Comment by immutableT** [comment]
@mikedanese 
Thank you for the review.
Squashing done.
